spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/AsyncRabbitTemplate.java

```diff
@@ -421,6 +475,11 @@ public void onMessage(Message message) {
 						((RabbitMessageFuture) future).set(message);
 					}
 				}
+				else {
+					if (logger.isDebugEnabled()) {
+						logger.debug("No pending reply - perhaps timed out");
```

Comment:
Wouldn't it be better to `warn()` instead?
Together with the `message`, of course, to be able diagnose it from logs. 
Something similar what we have in the `TemporaryReplyChannel`:
```
if (errorDescription != null) {
	if (logger.isWarnEnabled()) {
		logger.warn(errorDescription + ":" + message);
	}
	if (GenericMessagingTemplate.this.throwExceptionOnLateReply) {
		throw new MessageDeliveryException(message, errorDescription);
	}
}
```